bindings/nodejs/src/lib.rs

```diff
@@ -23,3 +36,100 @@ pub fn debug() -> String {
         .finish();
     format!("{:?}", op.metadata())
 }
+
+
+#[napi]
+pub struct OperatorFactory {}
+
+#[napi]
+impl OperatorFactory {
+    #[napi]
+    pub fn memory() -> napi::Result<OpendalStore> {
+        let op = Operator::create(services::Memory::default())
+            .unwrap()
+            .finish();
+
+        Ok(OpendalStore::new(op))
+    }
+}
+
+#[allow(dead_code)]
+#[napi]
+pub struct ObjectMeta {
+    pub location: String,
+    pub last_modified: i64,
+    pub size: u32
+}
+
+#[napi]
+pub struct OpendalStore {
+    inner: Operator
+}
+
+#[napi]
+impl OpendalStore {
+    pub fn new(op: Operator) -> Self {
+        Self { inner: op }
+    }
+
+    #[napi]
+    pub fn meta(&self) -> Result<String> {
+        Ok(format!("{:?}", self.inner.metadata()))
+    }
+
+    #[napi]
+    pub async fn head(&self, path: String) -> Result<ObjectMeta> {
```

Comment:
**request**: Please export the same API like OpenDAL does.

**question**: We can't use `&str` in napi?

Reply:
> We can't use `&str` in napi?

Yes. They did not implement for `&str`.